ecl/hqlcpp/hqlsource.cpp

```diff
@@ -2845,7 +2845,7 @@ class DiskReadBuilderBase : public SourceBuilder
 {
 public:
     DiskReadBuilderBase(HqlCppTranslator & _translator, IHqlExpression *_tableExpr, IHqlExpression *_nameExpr, bool canReadGenerically)
-        : SourceBuilder(_translator, _tableExpr, _nameExpr), monitors(_tableExpr, _translator, 0, true, false)
+        : SourceBuilder(_translator, _tableExpr, _nameExpr), monitors(_tableExpr, _translator, 0, true, true)
```

Comment:
I'm not sure why the monitors is being constructed with _createValueSets==true

Reply:
I had similarly questioned the change on line 79 of ecl/hqlcpp/hqlcfilter.cpp,
but this change is the answer. It is setting to true, because it used to have a condition to force on if isDiskRead, now it's explicit here.

Reply:
Why are we forcing use of valueSets on disk reads? Is the same option not also applicable to them?